platforms/common/px4_work_queue/WorkQueueManager.cpp

```diff
@@ -205,10 +205,19 @@ WorkQueueManagerRun(int, char **)
 			// stack size
 #ifndef __PX4_QURT
 			const size_t stacksize = math::max(PTHREAD_STACK_MIN, PX4_STACK_ADJUSTED(wq->stacksize));
+#   ifdef __PX4_POSIX
```

Comment:
How about flattening it out so that we only need a single call to pthread_attr_setstacksize and a single debug message?

``` C++

#if defined(__PX4_QURT)
	const size_t stacksize = math::max(8 * 1024, PX4_STACK_ADJUSTED(wq->stacksize));
#elif defined(__PX4_NUTTX)
	const size_t stacksize = math::max(PTHREAD_STACK_MIN, wq->stacksize);
#elif defined(__PX4_POSIX)
	const unsigned int page_size = sysconf(_SC_PAGESIZE);
	const size_t stacksize_adj = math::max(PTHREAD_STACK_MIN, PX4_STACK_ADJUSTED(wq->stacksize));
	const size_t stacksize = (stacksize_adj + page_size - (stacksize_adj % page_size));
#endif

int ret_setstacksize = pthread_attr_setstacksize(&attr, stacksize);

```

Reply:
This is a much cleaner approach. I like it.

Reply:
@dagar Do you voluntarily dropped the PX4_STACK_ADJUSTED in the nuttx case?

Reply:
Yes, but it's fine either way. The stacks are sized for NuttX, then scaled for the other platforms.

https://github.com/PX4/Firmware/blob/a08abccdd55dabba37c5df0eb5b62fddff3a5f47/platforms/common/include/px4_platform_common/posix.h#L127